cypress/integration/utils/sharedTests/testSaveQuery.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  newClickFlow,
+  newExpectation,
+  newExpectationWithClickFlows,
+  newShouldArgs,
+} from "../helpers";
+import { Tests } from "../types";
+import { mergingAndDeletingBranch } from "./testDocs";
+import { typingExpectation } from "./testIssues";
+
+const beVisible = newShouldArgs("be.visible");
+const notExist = newShouldArgs("not.exist");
+const beVisibleAndContain = (value: string) =>
+  newShouldArgs("be.visible.and.contain", value);
+const testQueryName = "test-query-name";
+const testQueryDescription = "test-query-description";
+
+const createPullRequest = () => [
+  newExpectationWithClickFlows(
+    "should show create pull request button",
+    "[data-cy=create-pull]",
+    beVisible,
+    [newClickFlow("[data-cy=create-pull]", [])],
+  ),
+  newExpectationWithClickFlows(
+    "should be able to create pull request",
+    "[data-cy=create-pull-button]",
+    beVisible,
+    [newClickFlow("[data-cy=create-pull-button]", [])],
+  ),
+];
+
+export const testSaveQuery: Tests = [
+  //! NAVIGATE TO THE DATABASE TAB
+  newExpectationWithClickFlows(
+    "should be able to navigate to database tab",
+    "[data-cy=repo-database-tab]",
+    beVisible,
+    [newClickFlow("[data-cy=repo-database-tab]", [])],
+  ),
+
+  //! SAVE THE QUERY
+  newExpectationWithClickFlows(
+    "should show save query button",
+    "[data-cy=save-query-button]",
+    beVisible,
+    [
+      newClickFlow(
+        "[data-cy=save-query-button]",
+        [
+          typingExpectation(testQueryName, "query-name"),
+          typingExpectation(testQueryDescription, "query-description"),
+        ],
+        "[data-cy=query-save-button]",
+      ),
+    ],
+  ),
+  newExpectation(
+    "should have saved query listed",
+    "[data-cy=workspace-commit-list]>li>div>a",
```

Comment:
I think this checks all `li` in the list. Would be better to check the first one? Like 
```suggestion
    "[data-cy=workspace-commit-list] > li:first a",
```
or something